hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -837,6 +850,42 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Reconcile the HostedControlPlane kubeadminPassword
+	if hcp.Status.KubeadminPassword != nil {
+		src := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: hcp.Namespace,
+				Name:      hcp.Status.KubeadminPassword.Name,
+			},
+		}
+		err := r.Client.Get(ctx, client.ObjectKeyFromObject(src), src)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get controlplane kubeadmin password secret %q: %w", client.ObjectKeyFromObject(src), err)
+		}
+		dest := manifests.KubeadminPasswordSecret(hcluster.Namespace, hcluster.Name)
+		_, err = createOrUpdate(ctx, r.Client, dest, func() error {
+			key := "password"
+			srcData, srcHasData := src.Data[key]
+			if !srcHasData {
```

Comment:
Should we really duplicate this rather than just copy whatever content is there over?

Reply:
makes sense to me

Reply:
updated